tests/integration/interface_common_test.py

```diff
@@ -34,3 +65,30 @@ def test_iface_description_removal(eth1_up):
     libnmstate.apply(desired_state)
     current_state = statelib.show_only(('eth1',))
     assert Interface.DESCRIPTION not in current_state[Interface.KEY][0]
+
+
+def test_take_over_virtual_interface_then_remove(ip_link_dummy):
+    dummy_desired_state = ip_link_dummy
+    assertlib.assert_state_match(dummy_desired_state)
+
+    dummy_desired_state[Interface.KEY][0][
+        Interface.STATE
```

Comment:
nit: Please use a temporary variable to make this more readable.

Reply:
Done